### DISCUSSION TITLE
Speed comparison : slower than whisper-jax ?

### DISCUSSION THREAD

**Discussion by Vultik** [SYSTRAN/faster-whisper#195] [Q&A]
Hi, 
How does `faster-whisper` compare with [https://huggingface.co/spaces/sanchit-gandhi/whisper-jax](https://huggingface.co/spaces/sanchit-gandhi/whisper-jax) ?
Using [JAX](https://github.com/google/jax) implementation it takes only 15s on large-v2 model to transcribe 10min of speech.
Is it possible to combine it with CTranslate2 ?

**Comment by guillaumekln** [comment]
Hi,

I haven't actually tested it but I expect whisper-jax to be faster on long files due to its batching strategy. For short files faster-whisper is faster according to some users: https://github.com/sanchit-gandhi/whisper-jax/issues/8.

However, whisper-jax does not implement the same transcription logic as openai/whisper so the transcription quality is probably reduced and the output less consistent.

If we implement a similar batching strategy here it's likely that we can match or even surpass the speed of whisper-jax on GPU. Note that apparently whisperX is planning to release a batch execution using faster-whisper so we'll see how that works: https://github.com/m-bain/whisperX/issues/159#issuecomment-1521619648

**Reply by Vultik** [reply]
Salut Guillaume,
Thanks for your quick response !
In fact, i have only tested whisper-jax and faster-whisper on big audio files. I will follow the batch execution made by m-bain :)

**Reply by toanhuynhnguyen** [reply]
@Vultik How did you test for the short files?
